roles/ceph-facts/tasks/facts.yml

```diff
@@ -318,6 +318,7 @@
     - groups.get(grafana_server_group_name, []) | length > 0
     - ip_version == 'ipv4'
     - dashboard_enabled | bool
+    - inventory_hostname in groups[grafana_server_group_name]
```

Comment:
in fact you don't need this condition (and the other one with `not in`)
In both scenarios (collocated or external) you need the grafana address from the grafana node. So the only way is to use `groups[grafana_server_group_name][0]` like you did in the external code but we can directly integrate that in the current code no need to duplicate the code.